src/__experimental__/components/date/date.component.js

```diff
@@ -179,17 +170,28 @@ class BaseDateInput extends React.Component {
     this.setState({ selectedDate: newDate });
   };
 
-  emitOnChangeCallback = (stringDate) => {
-    const changePayload = {
-      target: {
-        name: this.props.name,
-        value: stringDate
-      }
-    };
-
-    if (this.props.onChange) this.props.onChange(changePayload);
+  emitOnChangeCallback = (ev, isoFormattedValue) => {
+    if (this.props.onChange) {
+      const event = this.buildCustomTarget(ev, isoFormattedValue);
+      this.props.onChange(event);
+    }
   };
 
+  buildCustomTarget = (ev, isoFormattedValue) => {
```

Comment:
Maybe `buildCustomEvent` might be a better name? Or `customiseEventTarget` ?

(The name `buildCustomTarget` sounds like it returns a custom target object, whereas it actually returns a custom event object.)

Reply:
Yep agreed I had it as `buildCustomEvent` originally so will go with that

Reply:
Great, thanks :thumbsup: